packages/common/index.ts

```diff
@@ -1392,6 +1397,51 @@ export type sendEmailAsync = {
   subject: string
   type: asyncQuestionEventType
 }
+
+export class CourseSettingsResponse {
+  @IsInt()
+  courseId!: number
+
+  @IsBoolean()
+  chatBotEnabled!: boolean
+
+  @IsBoolean()
+  asyncQueueEnabled!: boolean
+
+  @IsBoolean()
+  adsEnabled!: boolean
+
+  @IsBoolean()
+  queueEnabled!: boolean
+
+  @IsOptional()
+  @IsBoolean()
+  settingsFound?: boolean = true //this is mostly just for debugging purposes by viewing network responses
+
+  constructor(init?: Partial<CourseSettingsResponse>) {
+    Object.assign(this, init)
+  }
+}
+
+const validFeatures = [
+  'chatBotEnabled',
+  'asyncQueueEnabled',
+  'adsEnabled',
+  'queueEnabled',
+]
+
+export class CourseSettingsRequestBody {
+  @IsBoolean()
+  value!: boolean
+
+  @IsIn(validFeatures)
+  feature!: string
+
+  static isValidFeature(feature: string): boolean {
```

Comment:
Methods are static by default?

Reply:
I asked chatgpt about this and it seems to disagree:

"A static method is a method that belongs to the class itself, not an instance of the class. You can call a static method without creating an instance of the class.

In your code, isValidFeature is a static method, which means you can call it as CourseSettingsRequestBody.isValidFeature(), without creating an instance of CourseSettingsRequestBody.

If you remove the static keyword, isValidFeature would become an instance method, and you would need to create an instance of CourseSettingsRequestBody to call it, like this:
```tsx
const requestBody = new CourseSettingsRequestBody();
requestBody.isValidFeature();
```
"